### ISSUE TITLE
Strict connection mode yaml not documented

### ISSUE THREAD

**Issue by wormiedk** [home-assistant/home-assistant.github.io#32479]
### Feedback

The parameters for strict connection mode are specified but not what the yaml entry is called. I guess it might be
strict_connection_mode: 

The documentation reads
```
http:
  use_x_forwarded_for: true
  trusted_proxies:
    - 10.0.0.200      # Add the IP address of the proxy server
    - 172.30.33.0/24  # You may also provide the subnet mask
   
**Strict connection mode**
The following modes are supported:

disabled: Strict connection mode is disabled, and all unauthenticated requests are allowed (Same as before 2024.5).
guard_page: Home Assistant will answer any unauthenticated requests with a guard page, informing the user about the strict connection mode.
drop_connection: Home Assistant will drop/close the connection for any unauthenticated requests, similar to when no Home Assistant instance is running.
```

### URL

https://rc.home-assistant.io/integrations/http/#strict-connection-mode

### Version

2024.5

**Comment by wormiedk** [comment]
Appaently not :) 

```
http:
  ip_ban_enabled: true
  login_attempts_threshold: 5
  use_x_forwarded_for: true
  strict_connection_mode: guard_page
```
Gives this error: 

Invalid config for 'http' at configuration.yaml, line 10: 'strict_connection_mode' is an invalid option for 'http', check: http->strict_connection_mode

**Comment by MrJelly17** [comment]
It needs to be 
strict_connection:

**Comment by wormiedk** [comment]
I just tried enabling it. I think the documentation needs to state how to login to the app from a new device.
I tried
1) Create a temporary token
2) Login to browser on device
3) Still no access to the app.
4) Log out of home asssistant app
5) Log into home assistant app while on local wifi
6) Home assistant app states that I have no access.
7) Disabled strict login :) 

Ideally I would be able to validate the login to a new device from a device on my local lan instead of jumping through hoops :)
